R/drive_search.R

```diff
@@ -53,11 +53,13 @@ drive_search <- function(pattern = NULL, type = NULL, ..., verbose = TRUE) {
   }
 
   if (!is.null(type)) {
-    mime_type <- drive_mime_type(type, verbose = verbose)
-    if (!is.null(mime_type)) {
-    params$q <- glue::collapse(c(params$q, paste0("mimeType = '", mime_type,"'")),
-                               sep = " and ")
-    }
+    mime_type <- drive_mime_type(type)
+    ## if they are all NA, this will error, because drive_mime_type
+    ## doesn't allow it, but if one is NA and the others aren't,
+    ## this will still pass the ok ones through.
+    params$q <- paste(c(params$q,
+                        paste0("mimeType = '", mime_type,"'", collapse = " or ")),
```

Comment:
Hmmm, I hadn't thought about multiple MIME types here. I think we should probably only allow one? Because with `q`, we are going to collapse multiple clauses with "and", so it feels weird to combine multiple `type`s with "or". If people want an "or" clause for mime types, they can build it themselves and provide via `q`.